File: 卡码/代码随想录/1/08-贪心算法/Java/greedy/src/main/java/com/kurumi/greedy/_02_中等/LeetCode_0055.java
//55. 跳跃游戏
package com.kurumi.greedy._02_中等;

public class LeetCode_0055 {

}
class Solution {
  public boolean canJump(int[] nums) {
    int cover=0;
    if(nums.length==1) return true;
    for(int i=0;i<=cover;i++){
      cover=Integer.max(i+nums[i], cover);
      if(cover>=nums.length-1) return true;
    }
    return false;
  }
}

File: 卡码/代码随想录/1/08-贪心算法/Python/02-中等/LeetCode_0055.py
#55. 跳跃游戏
from typing import *

class Solution:
  def canJump(self, nums: List[int]) -> bool:
    if len(nums)==1:
      return True
    cover:int=0
    index:int=0
    while index<=cover:
      cover=max(cover,index+nums[index])
      if cover>=len(nums)-1:
        return True
      index+=1
    return False